File: day02/day02.py
#!/usr/bin/python
from itertools import permutations


rows = [[int(y) for y in x.split()] for x in open('input.txt').readlines()]
print(sum(max(row) - min(row) for row in rows))
print(sum(a//b for row in rows for a, b in permutations(row, 2) if a % b == 0))

File: day02/day02.clj
(require '[clojure.string :as str])

(def rows (->> (str/split-lines (slurp "input.txt"))
           (map #(str/split % #"\t"))
           (map #(map read-string %))))

(prn (->> rows (map #(- (apply max %) (apply min %))) (apply +)))
(prn (->> rows (map #(first
                      (for [a % b % :when (> a b) :when (zero? (mod a b))]
                        (/ a b))))
          (apply +)))
